### ISSUE TITLE
Rejected orders should be added to order history

### ISSUE THREAD

**Issue by bdowling** [alpacahq/Alpaca-API#29]
Rejected orders should also be recorded in the order book, otherwise the onus is on the customer to log these failures, but it would be better to have a consolidated view in your account. 

Adding a `rejected_reason` attribute would also be needed.  (Trying to think if there are assorted cancel reasons as well, a more common attribute came could be used...)

**Comment by codebeaulieu** [comment]
Agreed, I'd rather a verbose order history that I can filter myself.
